bc_obps/service/data_access_service/report_service.py

```diff
@@ -0,0 +1,12 @@
+from uuid import UUID
+from reporting.models.report import Report
+
+
+class ReportDataAccessService:
+    @classmethod
+    def report_exists(cls, operation_id: UUID, reporting_year: int) -> Report | None:
+        return (
+            Report.objects.prefetch_related("operation")
+            .filter(operation__id=operation_id, reporting_year=reporting_year)
+            .first()
```

Comment:
We should use `.exists()` to align with the method name. Using `.exists()` ensures that we receive a boolean value instead of an empty QuerySet when the queried item does not exist.

Reply:
The intent here was to return the report if it exists, so that we can redirect the user directly. The `exists` in the name is only to say that it doesn't blow up if there is no report